src/Nfts/Nfts.tsx

```diff
@@ -0,0 +1,242 @@
+import React, {ReactNode} from 'react'
+import {defineMessages, useIntl} from 'react-intl'
+import {RefreshControl, ScrollView, StyleSheet, Text, View} from 'react-native'
+import {SafeAreaView} from 'react-native-safe-area-context'
+
+import {Icon, Spacer} from '../components'
+import {useFilteredNfts} from './hooks'
+import {ImageGallery, SkeletonGallery} from './ImageGallery'
+import {useNavigateTo} from './navigation'
+import {NoNftsScreen} from './NoNftsScreen'
+
+export const Nfts = () => {
+  const {search, nfts, isLoading, refetch, isRefetching, isError} = useFilteredNfts()
+  const navigateTo = useNavigateTo()
+  const navigateToDetails = (id: string) => navigateTo.nftDetails(id)
+  const handleNftSelect = (index: number) => navigateToDetails(nfts[index].id)
```

Comment:
too much abstraction
inline `onPress={() => navigateTo.details(nft.id)}`

Reply:
I think is more related to `navigateToDetails` that is used only inside the handler, maybe `handleNftSelect = (index: number) => navigateTo.nftDetails(nfts[index].id)`